src/store/transaction.ts

```diff
@@ -83,30 +83,40 @@ export const actions = actionTree(
   {
     async watchTransaction({ commit, dispatch, state }, { transactionHash, existingTransaction }) {
```

Comment:
Why there is an existingTransaction? This is something that can be checked by transactionHash ← sign that you're splitting the business-logic between multiple parts of the app which should be avoided.

Good architecture:

- define method to trigger watching some transaction
- define separate trigger to watch the balance
- accept either ZkInTx / TokenBalance or hash as a trigger
- inside the store check if there is a watcher already by this unique hash: if not — create one, if yes — ignore
- define some event that should be thrown when the transaction / balance is complete
- set a watcher to catch this event and drop overall watch of the particular object
- run some kind of check in lazy-mode when initializing the balance / transactions list etc.

@JackHamer09